cmd/minishift/cmd/profile/profile.go

```diff
@@ -19,13 +19,15 @@ package profile
 import (
 	"strings"
 
+	cmdUtil "github.com/minishift/minishift/cmd/minishift/cmd/util"
 	"github.com/minishift/minishift/pkg/util/os/atexit"
 	"github.com/spf13/cobra"
 )
 
 const (
 	emptyProfileMessage  = "A profile name must be provided. Run `minishift profile list` for a list of existing profiles."
 	extraArgumentMessage = "You have provided more arguments than required. You must provide a single profile name."
+	invalidNameMessage   = "Profile name only allow alphanumeric characters."
```

Comment:
"Profile name only allow alphanumeric characters." -> "Profile names must consist of alphanumeric characters only."

This does not match, however, the implementation, which is that the profile name is alphanumeric AND starts with at least one [:alpha:] character. 

We also need to update the docs and add at least a note there.